### PR TITLE
Added newTurnUnit and newTurnItem calls before the first player phase

### PR DESCRIPTION
**Pull request by memmaker**
Also added hasItemScript to BattleUnits

### CODE REVIEW

File: src/Savegame/BattleUnit.cpp [hunk 1]
```diff
@@ -5119,6 +5119,26 @@ void getFactionScript(const BattleUnit *bu, int &faction)
 	faction = 0;
 }
 
+void hasItemScript(BattleUnit* bu, int &result, ScriptText itemName, int count)
```
Comment (Yankes) [inline src/Savegame/BattleUnit.cpp]: I think this should go to different PR, for function itself.
If we iterate for whole inventory and do lot of work why not return item it self?
Second, I would prefer having "strong" typed API, aka pass `const RuleItem*` not string.
This would require new function in mod that will convert string to rule object, but in long run this will be more correct version.

Comment (memmaker) [reply src/Savegame/BattleUnit.cpp]: Seperate PR: Will be done.
Return the BattleItem itself: Agreed.
Changing the argument to RuleItem: How would I get the RuleItem reference I want to search for? The use case is checking if a unit has a specific item in the inventory.

Comment (Yankes) [reply src/Savegame/BattleUnit.cpp]: You miss: "new function in mod that will convert string to rule object", we should start allowing scripts to access arbitrary unit and item rules objects. This will be building block for future functionalities, like create new item or new unit.

Comment (memmaker) [reply src/Savegame/BattleUnit.cpp]: Ah, I see. Yeah, I didn't quite read that right. Apologies.

**Comment by Yankes** [comment]
Manually merged outside GitHub
